server/src/main/java/org/elasticsearch/health/HealthComponent.java

```diff
@@ -14,34 +14,36 @@
 import java.io.IOException;
 import java.util.Collection;
 import java.util.List;
+import java.util.Map;
 import java.util.TreeMap;
+import java.util.stream.Collectors;
 
 import static java.util.stream.Collectors.collectingAndThen;
 import static java.util.stream.Collectors.groupingBy;
 import static java.util.stream.Collectors.toList;
 
-public record HealthComponentResult(String name, HealthStatus status, List<HealthIndicatorResult> indicators) implements ToXContentObject {
+public record HealthComponent(String name, HealthStatus status, Map<String, HealthIndicator> indicators) implements ToXContentObject {
```

Comment:
I think we would like to keep the original name (`HealthComponentResult`) here. It is the result of a calculation, a "record" of the data more than the component (which we may need something for in the future) and aligns with `HealthIndicatorResult`.